app/controllers/schemes_controller.rb

```diff
@@ -51,17 +51,34 @@ def new_deactivation
   end
 
   def deactivate_confirm
-    @affected_logs = @scheme.lettings_logs.visible.after_date(params[:deactivation_date])
-    if @affected_logs.count.zero?
+    @deactivation_date = Time.zone.parse(params[:deactivation_date])
+    @affected_logs = @scheme.lettings_logs.visible.after_date(@deactivation_date)
+    @deactivation_date_type = params[:deactivation_date_type]
+
+    scheme_locations = @scheme.locations.confirmed
+
+    locations_active_on_deactivation_date, remaining_locations = scheme_locations.partition do |location|
+      location.status_at(@deactivation_date) == :active
+    end
+
+    locations_deactivating_after_deactivation_date, remaining_locations = remaining_locations.partition do |location|
+      location.status_at(@deactivation_date) == :deactivating_soon || location.status_at(@deactivation_date) == :reactivating_soon
+    end
+
+    locations_startdate_after_deactivation_date, = remaining_locations.partition do |location|
+      location.status_at(@deactivation_date) == :activating_soon
+    end
```

Comment:
Using partition, grabs the relevant locations for the three scenarios that could mean the location is affected. The last partition drops locations not relevant.

Reply:
Would select do the same thing without the empty variable instead of the last partition?